build_webdemo.sh

```diff
@@ -0,0 +1,26 @@
+script_path=$(realpath $(dirname $0))
+build_dir="build_webdemo"
+
+# interesting for Node.js: -sNODERAWFS=1, allows direct os filesystem access
+LDFLAGS=$(echo $(cat << EOF
+-sUSE_ZLIB=1
+-sINVOKE_RUN=0
+-sMODULARIZE=1
+-sEXPORTED_RUNTIME_METHODS=['FS','callMain']
+-sEXPORT_NAME='SoPlex'
+-sSINGLE_FILE
+-sALLOW_MEMORY_GROWTH=1
+--shell-file ${script_path}/src/soplex_webdemo_shell.html
+EOF
+))
+
+# compare https://stackoverflow.com/a/6481016
+physical_cores=$(grep ^cpu\\scores /proc/cpuinfo | uniq | awk '{print $4}')
+
```

Comment:
I think you can just run make -j and it will use whatever cores are available, right?

Reply:
`make -j` (without any core count) does some very nasty thing: it uses unlimited parallelism. So if will spawn compile processes for as many C files as are out there. Okay, maybe SoPlex is not affected by this so much as there are not so many files, but in the general case, it will use up all the resources (like CPU) and if one is out of luck even the RAM. Then things will start crashing. Also, compile times will be worse because of the many context switches. Therefore, `make -j` should always be used with a fixed core count. However, it is undesirable to have the entire system blocked during compilation, as one typically wants to keep browsing through the code - or the web. Therefore, it would be nice to use more than one core, but not all of them. The solution above uses all physical cores, which means that if the system does hyperthreading (1 physical core = 2 logical cores), this will use up half of the cores keeping the system responsive to user input _and_ reducing the compile time as the cache will be utilized more efficiently. I've seen other build files which have hard coded numbers like 4 or 8 in it. Would do as well. It is somehow Linux specific, indeed. Would not run on a Mac, for Windows there is WSL. So how about just fixing it to 8 and let the user increase it on demand?

Reply:
Yeah sure that makes sense. Unfortunately, since SoPlex is super header-heavy (because of all the templating) you don't gain too much by compiling in parallel anyway.